### PR TITLE
[PAPI-284] Amend Platform Hub to return StratisId

### PR DESCRIPTION
**Pull request by drmathias**
* Moved several objects to SSAS.NET package which is now referenced
* Replaces `GetEncryptedConnectionId` hub method with `GetStratisId` which is built with a 5 minute expiry
* Updated all mentions of SOAP to SSAS
* PlatformHub path changed to `/socket`

### CODE REVIEW

File: src/Opdex.Platform.WebApi/Validation/NetworkAddressValidator.cs [hunk 1]
```diff
@@ -17,6 +17,18 @@ public override bool IsValid(ValidationContext<T> context, Address value)
     protected override string GetDefaultMessageTemplate(string errorCode) => "{PropertyName} must be valid address.";
 }
 
+public class NetworkAddressStringValidator<T> : PropertyValidator<T, string>, INetworkAddressValidator
+{
+    public override string Name => "NetworkAddress";
+
+    public override bool IsValid(ValidationContext<T> context, string value)
+    {
+        return Address.TryParse(value, out var address) && address != Address.Empty && address != Address.Cirrus.ToString();
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/Validation/NetworkAddressValidator.cs:26]: depending on where this is used, CRS might be problematic

File: src/Opdex.Platform.WebApi/appsettings.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "Opdex": {
       "SigningKey": ""
     },
-    "StratisOpenAuthProtocol": {
+    "StratisSignatureAuthProtocol": {
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/appsettings.json:22]: Note: we will need to remember to change key vault values for dev/test/main

File: src/Opdex.Platform.WebApi/Validation/NetworkAddressValidator.cs (hunk 1 above)
Comment (drmathias) [reply src/Opdex.Platform.WebApi/Validation/NetworkAddressValidator.cs:26]: We'll create a separate validator `NetworkAddressOrCrsValidator` for that. Will remove the useless `.ToString()` here, just seen that.

File: src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
     /// </summary>
-    /// <returns>Encrypted connection id.</returns>
-    public string GetEncryptedConnectionId()
+    /// <returns>Stratis ID.</returns>
+    public string GetStratisId()
     {
         var encryptedConnectionId = _twoWayEncryptionProvider.Encrypt(Context.ConnectionId);
-        return Base64Extensions.UrlSafeBase64Encode(encryptedConnectionId);
+        var uid = Base64Extensions.UrlSafeBase64Encode(encryptedConnectionId);
+        var stratisId = new StratisId(_authCallback, uid, DateTimeOffset.UtcNow.AddMinutes(5).ToUnixTimeSeconds());
```
Comment (mrtpain) [inline src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs:31]: Will this prefix with `sid:`?

**Review by mrtpain — APPROVED** [review]
lgtm, one question about the `sid:` prefix.

We can keep this one open until the frontend has a matching pr to use `/socket` instead

File: src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs (hunk 3 above)
Comment (drmathias) [reply src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs:31]: `stratisId.ToString()` builds the full URI with the prefix

Comment (mrtpain) [reply src/Opdex.Platform.Infrastructure/Clients/SignalR/PlatformHub.cs:31]: perfect

**Pull request merged**